<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.3.9: http://docutils.sourceforge.net/" />
<title>HELP</title>
<link rel="stylesheet" href="/static/help.css" type="text/css" />
</head>
<body>
<div class="toolHelp">
<div class="toolHelpBody">
<p><span style="color:black;font-size:xx-large"><strong>Cufflinks Overview</strong></span></p>
<p><a class="reference external" href="http://cufflinks.cbcb.umd.edu/">Cufflinks</a> assembles transcripts, estimates their abundances, and tests for differential expression and regulation in RNA-Seq samples. It accepts aligned RNA-Seq reads and assembles the alignments into a parsimonious set of transcripts. Cufflinks then estimates the relative abundances of these transcripts based on how many reads support each one.  Please cite: Trapnell C, Williams BA, Pertea G, Mortazavi AM, Kwan G, van Baren MJ, Salzberg SL, Wold B, Pachter L. Transcript assembly and abundance estimation from RNA-Seq reveals thousands of new transcripts and switching among isoforms. Nature Biotechnology doi:10.1038/nbt.1621</p>
<hr class="docutils">
<p><strong>Input formats</strong></p>
<p>Cufflinks takes a text file of SAM alignments as input. The RNA-Seq read mapper TopHat produces output in this format, and is recommended for use with Cufflinks. However Cufflinks will accept SAM alignments generated by any read mapper. Here's an example of an alignment Cufflinks will accept:</p>
<pre class="literal-block">s6.25mer.txt-913508   16      chr1 4482736 255 14M431N11M * 0 0 \
CAAGATGCTAGGCAAGTCTTGGAAG IIIIIIIIIIIIIIIIIIIIIIIII NM:i:0 XS:A:-
</pre>
<p>Note the use of the custom tag XS. This attribute, which must have a value of "+" or "-", indicates which strand the RNA that produced this read came from. While this tag can be applied to any alignment, including unspliced ones, it must be present for all spliced alignment records (those with a 'N' operation in the CIGAR string).
The SAM file supplied to Cufflinks must be sorted by reference position. If you aligned your reads with TopHat, your alignments will be properly sorted already. If you used another tool, you may want to make sure they are properly sorted as follows:</p>
<pre class="literal-block">sort -k 3,3 -k 4,4n hits.sam &gt; hits.sam.sorted
</pre>
<p>NOTE: Cufflinks currently only supports SAM alignments with the CIGAR match ('M') and reference skip ('N') operations. Support for the other operations, such as insertions, deletions, and clipping, will be added in the future.</p>
<hr class="docutils">
<p><strong>Outputs</strong></p>
<p>Cufflinks produces three output files:</p>
<p>Transcripts and Genes:</p>
<p>This GTF file contains Cufflinks' assembled isoforms. The first 7 columns are standard GTF, and the last column contains attributes, some of which are also standardized (e.g. gene_id, transcript_id). There one GTF record per row, and each record represents either a transcript or an exon within a transcript. The columns are defined as follows:</p>
<pre class="literal-block">Column number   Column name   Example     Description
-----------------------------------------------------
1               seqname       chrX        Chromosome or contig name
2               source        Cufflinks   The name of the program that generated this file (always 'Cufflinks')
3               feature       exon        The type of record (always either "transcript" or "exon").
4               start         77696957    The leftmost coordinate of this record (where 0 is the leftmost possible coordinate)
5               end           77712009    The rightmost coordinate of this record, inclusive.
6               score         77712009    The most abundant isoform for each gene is assigned a score of 1000. Minor isoforms are scored by the ratio (minor FPKM/major FPKM)
7               strand        +           Cufflinks' guess for which strand the isoform came from. Always one of '+', '-' '.'
7               frame         .           Cufflinks does not predict where the start and stop codons (if any) are located within each transcript, so this field is not used.
8               attributes    See below
</pre>
<p>Each GTF record is decorated with the following attributes:</p>
<pre class="literal-block">Attribute       Example       Description
-----------------------------------------
gene_id         CUFF.1        Cufflinks gene id
transcript_id   CUFF.1.1      Cufflinks transcript id
FPKM            101.267       Isoform-level relative abundance in Reads Per Kilobase of exon model per Million mapped reads
frac            0.7647        Reserved. Please ignore, as this attribute may be deprecated in the future
conf_lo         0.07          Lower bound of the 95% confidence interval of the abundance of this isoform, as a fraction of the isoform abundance. That is, lower bound = FPKM * (1.0 - conf_lo)
conf_hi         0.1102        Upper bound of the 95% confidence interval of the abundance of this isoform, as a fraction of the isoform abundance. That is, upper bound = FPKM * (1.0 + conf_lo)
cov             100.765       Estimate for the absolute depth of read coverage across the whole transcript
</pre>
<dl class="docutils">
<dt>Transcripts only:</dt>
<dd>This file is simply a tab delimited file containing one row per transcript and with columns containing the attributes above. There are a few additional attributes not in the table above, but these are reserved for debugging, and may change or disappear in the future.</dd>
</dl>
<p>Genes only:
This file contains gene-level coordinates and expression values.</p>
<hr class="docutils">
<p><strong>Cufflinks settings</strong></p>
<p>All of the options have a default value. You can change any of them. Most of the options in Cufflinks have been implemented here.</p>
<hr class="docutils">
<p><strong>Cufflinks parameter list</strong></p>
<p>This is a list of implemented Cufflinks options:</p>
<pre class="literal-block">-m INT    This is the expected (mean) inner distance between mate pairs. For, example, for paired end runs with fragments selected at 300bp, where each end is 50bp, you should set -r to be 200. The default is 45bp.
-s INT    The standard deviation for the distribution on inner distances between mate pairs. The default is 20bp.
-I INT    The minimum intron length. Cufflinks will not report transcripts with introns longer than this, and will ignore SAM alignments with REF_SKIP CIGAR operations longer than this. The default is 300,000.
-F        After calculating isoform abundance for a gene, Cufflinks filters out transcripts that it believes are very low abundance, because isoforms expressed at extremely low levels often cannot reliably be assembled, and may even be artifacts of incompletely spliced precursors of processed transcripts. This parameter is also used to filter out introns that have far fewer spliced alignments supporting them. The default is 0.05, or 5% of the most abundant isoform (the major isoform) of the gene.
-j        Some RNA-Seq protocols produce a significant amount of reads that originate from incompletely spliced transcripts, and these reads can confound the assembly of fully spliced mRNAs. Cufflinks uses this parameter to filter out alignments that lie within the intronic intervals implied by the spliced alignments. The minimum depth of coverage in the intronic region covered by the alignment is divided by the number of spliced reads, and if the result is lower than this parameter value, the intronic alignments are ignored. The default is 5%.
-G        Tells Cufflinks to use the supplied reference annotation to estimate isoform expression. It will not assemble novel transcripts, and the program will ignore alignments not structurally compatible with any reference transcript.
-N        With this option, Cufflinks excludes the contribution of the top 25 percent most highly expressed genes from the number of mapped fragments used in the FPKM denominator. This can improve robustness of differential expression calls for less abundant genes and transcripts.
</pre>

        </div>
        </div>
<div class="toolHelpBody">
            
<p><span style="color:black;font-size:xx-large"><strong>Tophat Overview</strong></span></p>
<p><a class="reference external" href="http://tophat.cbcb.umd.edu/">TopHat</a> is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons. Please cite: Trapnell, C., Pachter, L. and Salzberg, S.L. TopHat: discovering splice junctions with RNA-Seq. Bioinformatics 25, 1105-1111 (2009).</p>
<hr class="docutils">
<p><strong>Input formats</strong></p>
<p>Tophat accepts files in Sanger FASTQ format. Use the FASTQ Groomer to prepare your files.</p>
<hr class="docutils">
<p><strong>Outputs</strong></p>
<p>Tophat produces two output files:</p>
<ul class="simple">
<li>junctions -- A UCSC <a class="reference external" href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> track of junctions reported by TopHat. Each junction consists of two connected BED blocks, where each block is as long as the maximal overhang of any read spanning the junction. The score is the number of alignments spanning the junction.</li>
<li>accepted_hits -- A list of read alignments in <a class="reference external" href="http://samtools.sourceforge.net/">BAM</a> format.</li>
</ul>
<p>Two other possible outputs, depending on the options you choose, are insertions and deletions, both of which are in BED format.</p>
<hr class="docutils">
<p><strong>Tophat settings</strong></p>
<p>All of the options have a default value. You can change any of them. Some of the options in Tophat have been implemented here.</p>
<hr class="docutils">
<p><strong>Tophat parameter list</strong></p>
<p>This is a list of implemented Tophat options:</p>
<pre class="literal-block">-r                                This is the expected (mean) inner distance between mate pairs. For, example, for paired end runs with fragments
                                  selected at 300bp, where each end is 50bp, you should set -r to be 200. There is no default, and this parameter
                                  is required for paired end runs.
--mate-std-dev INT                The standard deviation for the distribution on inner distances between mate pairs. The default is 20bp.
-a/--min-anchor-length INT        The "anchor length". TopHat will report junctions spanned by reads with at least this many bases on each side of the junction. Note that individual spliced
                                  alignments may span a junction with fewer than this many bases on one side. However, every junction involved in spliced alignments is supported by at least one
                                  read with this many bases on each side. This must be at least 3 and the default is 8.
-m/--splice-mismatches INT        The maximum number of mismatches that may appear in the "anchor" region of a spliced alignment. The default is 0.
-i/--min-intron-length INT        The minimum intron length. TopHat will ignore donor/acceptor pairs closer than this many bases apart. The default is 70.
-I/--max-intron-length INT        The maximum intron length. When searching for junctions ab initio, TopHat will ignore donor/acceptor pairs farther than this many bases apart, except when such a pair is supported by a split segment alignment of a long read. The default is 500000.
-F/--min-isoform-fraction 0.0-1.0 TopHat filters out junctions supported by too few alignments. Suppose a junction spanning two exons, is supported by S reads. Let the average depth of coverage of
                                  exon A be D, and assume that it is higher than B. If S / D is less than the minimum isoform fraction, the junction is not reported. A value of zero disables the
                                  filter. The default is 0.15.
-g/--max-multihits INT            Instructs TopHat to allow up to this many alignments to the reference for a given read, and suppresses all alignments for reads with more than this many
                                  alignments. The default is 40.
-G/--GTF [GTF 2.2 file]           Supply TopHat with a list of gene model annotations. TopHat will use the exon records in this file to build a set of known splice junctions for each gene, and will attempt to align reads to these junctions even if they would not normally be covered by the initial mapping.
-j/--raw-juncs [juncs file]       Supply TopHat with a list of raw junctions. Junctions are specified one per line, in a tab-delimited format. Records look like: [chrom] [left] [right] [+/-], left and right are zero-based coordinates, and specify the last character of the left sequenced to be spliced to the first character of the right sequence, inclusive.
-no-novel-juncs                   Only look for junctions indicated in the supplied GFF file. (ignored without -G)
--no-closure-search               Disables the mate pair closure-based search for junctions. Currently, has no effect - closure search is off by default.
--closure-search                  Enables the mate pair closure-based search for junctions. Closure-based search should only be used when the expected inner distance between mates is small (about or less than 50bp)
--no-coverage-search              Disables the coverage based search for junctions.
--coverage-search                 Enables the coverage based search for junctions. Use when coverage search is disabled by default (such as for reads 75bp or longer), for maximum sensitivity.
--microexon-search                With this option, the pipeline will attempt to find alignments incident to microexons. Works only for reads 50bp or longer.
--butterfly-search                TopHat will use a slower but potentially more sensitive algorithm to find junctions in addition to its standard search. Consider using this if you expect that your experiment produced a lot of reads from pre-mRNA, that fall within the introns of your transcripts.
--segment-mismatches              Read segments are mapped independently, allowing up to this many mismatches in each segment alignment. The default is 2.
--segment-length                  Each read is cut up into segments, each at least this long. These segments are mapped independently. The default is 25.
--min-closure-exon                During closure search for paired end reads, exonic hops in the potential splice graph must be at least this long. The default is 50.
--min-closure-intron              The minimum intron length that may be found during closure search. The default is 50.
--max-closure-intron              The maximum intron length that may be found during closure search. The default is 5000.
--min-coverage-intron             The minimum intron length that may be found during coverage search. The default is 50.
--max-coverage-intron             The maximum intron length that may be found during coverage search. The default is 20000.
--min-segment-intron              The minimum intron length that may be found during split-segment search. The default is 50.
--max-segment-intron              The maximum intron length that may be found during split-segment search. The default is 500000.
</pre>

        </div>

<div class="toolHelpBody">
<p><span style="color:black;font-size:xx-large"><strong>Bowtie Overview</strong></span></p>            
<p><a class="reference external" href="http://bowtie-bio.sourceforge.net/index.shtml">Bowtie</a> is a short read aligner designed to be ultrafast and memory-efficient. It is developed by Ben Langmead and Cole Trapnell. Please cite: Langmead B, Trapnell C, Pop M, Salzberg SL. Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. Genome Biology 10:R25.</p>
<blockquote>
</blockquote>
<hr class="docutils">
<p><strong>Input formats</strong></p>
<p>Bowtie accepts files in Sanger FASTQ format. Use the FASTQ Groomer to prepare your files.</p>
<hr class="docutils">
<p><strong>Outputs</strong></p>
<p>The output is in SAM format, and has the following columns:</p>
<pre class="literal-block">  Column  Description
--------  --------------------------------------------------------
 1 QNAME  Query (pair) NAME
 2 FLAG   bitwise FLAG
 3 RNAME  Reference sequence NAME
 4 POS    1-based leftmost POSition/coordinate of clipped sequence
 5 MAPQ   MAPping Quality (Phred-scaled)
 6 CIGAR  extended CIGAR string
 7 MRNM   Mate Reference sequence NaMe ('=' if same as RNAME)
 8 MPOS   1-based Mate POSition
 9 ISIZE  Inferred insert SIZE
10 SEQ    query SEQuence on the same strand as the reference
11 QUAL   query QUALity (ASCII-33 gives the Phred base quality)
12 OPT    variable OPTional fields in the format TAG:VTYPE:VALUE
</pre>
<p>The flags are as follows:</p>
<pre class="literal-block">  Flag  Description
------  -------------------------------------
0x0001  the read is paired in sequencing
0x0002  the read is mapped in a proper pair
0x0004  the query sequence itself is unmapped
0x0008  the mate is unmapped
0x0010  strand of the query (1 for reverse)
0x0020  strand of the mate
0x0040  the read is the first read in a pair
0x0080  the read is the second read in a pair
0x0100  the alignment is not primary
</pre>
<p>It looks like this (scroll sideways to see the entire example):</p>
<pre class="literal-block">QNAME FLAG    RNAME   POS     MAPQ    CIAGR   MRNM    MPOS    ISIZE   SEQ     QUAL    OPT
HWI-EAS91_1_30788AAXX:1:1:1761:343    4       *       0       0       *       *       0       0       AAAAAAANNAAAAAAAAAAAAAAAAAAAAAAAAAAACNNANNGAGTNGNNNNNNNGCTTCCCACAGNNCTGG        hhhhhhh;;hhhhhhhhhhh^hOhhhhghhhfhhhgh;;h;;hhhh;h;;;;;;;hhhhhhghhhh;;Phhh
HWI-EAS91_1_30788AAXX:1:1:1578:331    4       *       0       0       *       *       0       0       GTATAGANNAATAAGAAAAAAAAAAATGAAGACTTTCNNANNTCTGNANNNNNNNTCTTTTTTCAGNNGTAG        hhhhhhh;;hhhhhhhhhhhhhhhhhhhhhhhhhhhh;;h;;hhhh;h;;;;;;;hhhhhhhhhhh;;hhVh
</pre>
<hr class="docutils">
<h1><strong>Bowtie settings</strong></h1>
<p>All of the options have a default value. You can change any of them. Most of the options in Bowtie have been implemented here.</p>
<hr class="docutils">
<p><strong>Bowtie parameter list</strong></p>
<p>This is an exhaustive list of Bowtie options:</p>
<p>For indexing (bowtie-build):</p>
<pre class="literal-block">-a                 No auto behavior. Disable the default behavior where bowtie automatically
                   selects values for --bmax/--bmaxdivn/--dcv/--packed parameters according
                   to the memory available. [off]
--packed           Packing. Use a packed representation for DNA strings. [auto]
--bmax INT         Suffix maximum. The maximum number of suffixes allowed in a block. [auto]
--bmaxdivn INT     Suffix maximum fraction. The maximum number of suffixes allowed in a block
                   expressed as a fraction of the length of the reference. [4]
--dcv INT          Difference-cover sample. Use INT as the period for the difference-cover
                   sample. [1024]
--nodc INT         No difference-cover sample. Disable the difference-cover sample. [off]
-r                 No reference indexes. Do not build the NAME.3.ebwt and NAME.4.ebwt portions
                   of the index. Used only for paired-end alignment. [off]
-o                 Offrate. How many Burrows-Wheeler rows get marked by the indexer. The
                   indexer will mark every 2^INT rows. The marked rows correspond to rows on
                   the genome. [5]
-t INT             Ftab. The lookup table used to calculate an initial Burrows-Wheeler range
                   with respect to the first INT characters of the query. Ftab is 4^INT+1
                   bytes. [10]
--ntoa             N conversion. Convert Ns to As before building the index. Otherwise, Ns are
                   simply excluded from the index and Bowtie will not find alignments that
                   overlap them. [off]
--big              Endianness. Endianness to use when serializing integers to the index file. [off]
--little           Endianness. [--little]
--seed INT         Random seed. Use INT as the seed for the pseudo-random number generator. [off]
--cutoff INT       Cutoff. Index only the first INT bases of the reference sequences (cumulative
                   across sequences) and ignore the rest. [off]
</pre>
<p>For aligning (bowtie):</p>
<pre class="literal-block">-s INT             Skip. Do not align the first INT reads or pairs in the input. [off]
-u INT             Align limit. Only align the first INT reads/pairs from the input. [no limit]
-5 INT             High-quality trim. Trim INT bases from the high-quality (left) end of each
                   read before alignment. [0]
-3 INT             Low-quality trim. Trim INT bases from the low-quality (right) end of each
                   read before alignment. [0]
-n INT             Mismatch seed. Maximum number of mismatches permitted in the seed (defined
                   with seed length option). Can be 0, 1, 2, or 3. [2]
-e INT             Mismatch quality. Maximum permitted total of quality values at mismatched
                   read positions. Bowtie rounds quality values to the nearest 10 and saturates
                   at 30. [70]
-l INT             Seed length. The number of bases on the high-quality end of the read to
                   which the -n ceiling applies. Must be at least 5. [28]
--nomaqround       Suppress MAQ rounding. Values are internally rounded to the nearest 10 and
                   saturate at 30. This options turns off that rounding. [off]
-v INT             MAQ- or SOAP-like alignment policy. This option turns off the default
                   MAQ-like alignment policy in favor of a SOAP-like one. End-to-end alignments
                   with at most INT mismatches. [off]
-I INT             Minimum insert. The minimum insert size for valid paired-end alignments.
                   Does checking on untrimmed reads if -5 or -3 is used. [0]
-X INT             Maximum insert. The maximum insert size for valid paired-end alignments.
                   Does checking on untrimmed reads if -5 or -3 is used. [250]
--fr               Mate orientation. The upstream/downstream mate orientations for a valid
                   paired-end alignment against the forward reference strand. [--fr]
--rf               Mate orientation. [off]
--ff               Mate orientation. [off]
--pairtries INT    Maximum alignment attempts for paired-end data. [100]
--nofw             No forward aligning. Choosing this option means that Bowtie will not attempt
                   to align against the forward reference strand. [off]
--norc             No reverse-complement aligning. Setting this will mean that Bowtie will not
                   attempt to align against the reverse-complement reference strand. [off]
--un FILENAME      Write all reads that could not be aligned to file [off]
--max FILENAME     Write all reads with a number of valid alignments exceeding the limit
                   set with the -m option to file [off]
--maxbts INT       Maximum backtracks. The maximum number of backtracks permitted when aligning
                   a read in -n 2 or -n 3 mode. [125 without --best] [800 with --best]
-y                 Try hard. Try as hard as possible to find valid alignments when they exist,
                   including paired-end alignments. [off]
--chunkmbs INT     Thread memory. The number of megabytes of memory a given thread is given to
                   store path descriptors in --best mode. [32]
-k INT             Valid alignments. The number of valid alignments per read or pair. [off]
-a                 All valid alignments. Choosing this means that all valid alignments per read
                   or pair will be reported. [off]
-m INT             Suppress alignments. Suppress all alignments for a particular read or pair
                   if more than INT reportable alignments exist for it. [no limit]
--best             Best mode. Make Bowtie guarantee that reported singleton alignments are
                   "best" in terms of stratum (the number of mismatches) and quality values at
                   mismatched position. [off]
--strata           Best strata. When running in best mode, report alignments that fall into the
                   best stratum if there are ones falling into more than one. [off]
-o INT             Offrate override. Override the offrate of the index with INT. Some row
                   markings are discarded when index read into memory. INT must be greater than
                   the value used to build the index (default: 5). [off]
--seed INT         Random seed. Use INT as the seed for the pseudo-random number generator. [off]
--snpphred INT     Use INT as the SNP penalty for decoding colorspace alignments. True ratio of
                   SNPs per base in the subject genome. [see --snpfrac]
--snpfrac DEC      Use DEC as the estimated ratio of SNPs per base when decoding colorspace
                   alignments. [0.001]
--col-keepends     Keep the extreme-end nucleotides and qualities when decoding colorspace
                   alignments. [off]
</pre>

        </div>
        
        
</body>
</html>
